api/api.go

```diff
@@ -461,6 +523,159 @@ func CreateOrb(ctx context.Context, logger *logger.Logger, namespace string, nam
 	return orb, err
 }
 
+func incOrbVersion(orbVersion string, segment string) (string, error) {
+	v, err := semver.NewVersion(orbVersion)
+	if err != nil {
+		return "", err
+	}
+
+	var v2 semver.Version
+	switch segment {
+	case "major":
+		v2 = v.IncMajor()
+	case "minor":
+		v2 = v.IncMinor()
+	case "patch":
+		v2 = v.IncPatch()
+	}
+
+	return v2.String(), nil
+}
+
+// IncrementOrb accepts an orb and segment to increment the orb.
+func IncrementOrb(ctx context.Context, logger *logger.Logger, configPath string, namespace string, orb string, segment string) (string, error) {
+	name := namespace + "/" + orb
+	id, err := getOrbID(ctx, logger, name)
+	if err != nil {
+		return "", err
+	}
+
+	v, err := OrbVersion(ctx, logger, namespace, orb)
+	if err != nil {
+		return "", err
+	}
+
+	v2, err := incOrbVersion(v, segment)
+	if err != nil {
+		return "", err
+	}
+
+	response, err := OrbPublishID(ctx, logger, configPath, id, v2)
+	if err != nil {
+		return "", err
+	}
+
+	if len(response.Errors) > 0 {
+		return "", response.ToError()
+	}
+
+	logger.Debug("Bumped %s#%s from %s by %s to %s\n.", name, id, v, segment, v2)
+
+	return v2, nil
+}
+
+// OrbVersion finds the latest published version of an orb and returns it.
+func OrbVersion(ctx context.Context, logger *logger.Logger, namespace string, orb string) (string, error) {
+	name := namespace + "/" + orb
+
+	var response struct {
+		Orb struct {
+			Versions []struct {
+				Version string
+			}
+		}
+	}
+
+	query := `query($name: String!) {
+			    orb(name: $name) {
+			      versions(count: 1) {
```

Comment:
I don't know much about graphQL, but from this query I can't tell that it's fetching latest. I guess the api-server always sorts them by semvar?

Reply:
I'm doing the simple thing here, that `orb source` was doing.

Presumably that returns the last created revision, though this may not be the `latest` version per say.

For example, if a user has published "1.0.0" then creates a "0.9.8", whichever was created last gets picked afaict.

Without changing the return value from the server, my best guess at a better implementation of this is to fetch all versions and sort them using one of the semver libraries we have available. Though this may be a little expensive..?